exporter/exporter.go

```diff
@@ -648,8 +648,10 @@ func (e *Exporter) scrapeRedisHost(ch chan<- prometheus.Metric) error {
 	}
 
 	if len(e.options.LuaScript) > 0 {
-		if err := e.extractLuaScriptMetrics(ch, c); err != nil {
-			return err
+		for _, script := range e.options.LuaScript {
+			if err := e.extractLuaScriptMetrics(ch, c, script); err != nil {
```

Comment:
My apologies, I should have caught this during the initial, cursory review but missed it.
I think you should continue here when an error occurs and 1) log the error to the console and b) record a gauge metric with success==1/error==0 with the script name as a label.

To do that you'll need to have the label so change LuaScript in the exporter struct to be an array of Filename string & Script []byte so you have the filename available.

You should probably also record that filename label with the lua script results in lua.go:30 to avoid collisions of two scripts using the same  variable name when returning results and for added info where a value is coming from.

Reply:
Hi,
Thank you for review. I will try to update it this week.

But are you sure that we want to start exporter even if file is not found?

> 1) log the error to the console
It's already doing it failing on first not accessible file `FATA[0000] Error loading script file test1.lua    err: open test1.lua: no such file or directory`

> You should probably also record that filename label with the lua script results in lua.go:30 to avoid collisions of two scripts using the same variable name when returning results and for added info where a value is coming from.

Yep, this sounds reasonable.

Reply:
You're right, this should fail fast on startup if a file is not present and the metric can record if executing the lua script succeeded or not.

Reply:
Hi @oliver006 
Merry Christmas and a happy new year!🎄 

I managed to check my PR during holidays and I think I implemented all we discussed.

Reply:
Thanks for all your effort here!
I'll merge this as soon as the CI server runs (it sometimes gets backed up)

Any merry Christmas and a happy, healthy new year to you too!

Reply:
Hi @oliver006
It seems CI has some issues?

Reply:
Yeah, the CI has issues, it's pretty annoying. Let me find time the next few days and migrate it to a new CI instance. In the meantime, I'll merge the PR.

Reply:
Finally got around to fixing the CI, your feature is released now as https://github.com/oliver006/redis_exporter/releases/tag/v1.46.0